clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -4192,6 +4193,142 @@ static bool ContainsWrapperAction(const Action *A) {
   return false;
 }
 
+// Put together an external compiler compilation call which is used instead
+// of the clang invocation for the host compile of an offload compilation.
+void Clang::ConstructHostCompilerJob(Compilation &C, const JobAction &JA,
+                                     const InputInfo &Output,
+                                     const InputInfoList &Inputs,
+                                     const llvm::opt::ArgList &TCArgs,
+                                     const Tool *T) const {
+
+  // The Host compilation step that occurs here is constructed based on the
+  // input from the user.  This consists of the compiler to call and the
+  // options that will be used during the compilation.
+  ArgStringList HostCompileArgs;
+  const InputInfo &InputFile = Inputs.front();
+  const auto &TC = getToolChain();
+
+  // Input file.
+  HostCompileArgs.push_back(InputFile.getFilename());
+
+  // When performing the host compilation, we are expecting to only be
+  // creating intermediate files.  These being preprocessed, assembly or
+  // object files.
+  // We are making assumptions in regards to what options are used to
+  // generate these intermediate files.
+  //                gcc/g++/clang/clang++/default | cl
+  //  Object:                   -c                | -c
+  //  Preprocessed:             -E                | -P -Fi<file>
+  //  Assembly:                 -S                | -c -Fa<file>
+  //  Header Input:        -include <file>        | -FI <file>
+  //
+  // The options used are determined by the compiler name and target triple.
+  auto *HostArg = TCArgs.getLastArg(options::OPT_fsycl_host_compiler_EQ);
```

Comment:
`HostArg` is somewhat confusing
```suggestion
  auto *HostCompilerDefArg = TCArgs.getLastArg(options::OPT_fsycl_host_compiler_EQ);
```